torch_em/shallow2deep/prepare_shallow2deep.py

```diff
@@ -504,11 +507,86 @@ def random_points(
     return features[samples], labels[samples]
 
 
+def worst_tiles(
+    features, labels, rf_id,
+    forests, forests_per_stage,
+    sample_fraction_per_stage,
+    img_shape,
+    tiles_shape=[51, 51],
```

Comment:
To be honest, I don't have a real good metric for that. Just tried with 7, 25, and 51 and the DiceLoss was smallest for 51. But you already mentioned that that's not a good proxy. So I guess we can use something smaller. What would you suggest?
With 51 usually 2-3 maxima were sampled :)

Reply:
with a patch size between 256 and 512

Reply:
> With 51 usually 2-3 maxima were sampled :)

But did it actually then sample points from all of them after class balancing?

Reply:
Also, now that I think about the class balancing more here: would it maybe make sense to compute the maxima for the classes independently? I think we might run into cases where we only sample one class otherwise.

Reply:
> > With 51 usually 2-3 maxima were sampled :)
> 
> But did it actually then sample points from all of them after class balancing?

at least one class did sample from all of them, but you are right for the others that's not guaranteed

Reply:
> Also, now that I think about the class balancing more here: would it maybe make sense to compute the maxima for the classes independently? I think we might run into cases where we only sample one class otherwise.

True, this could happen. I'll try to adapt it :)

Reply:
but then we loose a bit of the "a human would label localy" idea. because locality is then just guaranteed in a class but not over classes. But I think this still makes sense.